### ISSUE TITLE
TypeError on production server (ERR_INVALID_ARG_TYPE)

### ISSUE THREAD

**Issue by futurepastori** [vercel/next.js#31471]
### What version of Next.js are you using?

12.0.4

### What version of Node.js are you using?

14.17.6

### What browser are you using?

Chrome (it doesn't matter)

### What operating system are you using?

macOS

### How are you deploying your application?

elastic beanstalk

### Describe the Bug

Running `next start` after a successful build, the following error happens. Only happens on server-side rendered pages, as client-side rendered pages and navigation work as usual.

```
TypeError [ERR_INVALID_ARG_TYPE]: The "chunk" argument must be of type string or an instance of Buffer or Uint8Array. Received undefined
    at Writable.write (internal/streams/writable.js:285:13)
    at /Users/victor/Downloads/app-b64f-211115_222948-stage-211115_222948/node_modules/next/dist/server/render.js:1123:37
    at Array.forEach (<anonymous>)
    at /Users/victor/Downloads/app-b64f-211115_222948-stage-211115_222948/node_modules/next/dist/server/render.js:1123:16
    at /Users/victor/Downloads/app-b64f-211115_222948-stage-211115_222948/node_modules/next/dist/server/render.js:1110:13
    at /Users/victor/Downloads/app-b64f-211115_222948-stage-211115_222948/node_modules/next/dist/server/render.js:1110:47
    at /Users/victor/Downloads/app-b64f-211115_222948-stage-211115_222948/node_modules/next/dist/server/render.js:1128:9
    at /Users/victor/Downloads/app-b64f-211115_222948-stage-211115_222948/node_modules/next/dist/server/render.js:1110:13
    at /Users/victor/Downloads/app-b64f-211115_222948-stage-211115_222948/node_modules/next/dist/server/render.js:1110:47
    at /Users/victor/Downloads/app-b64f-211115_222948-stage-211115_222948/node_modules/next/dist/server/render.js:1128:9 {
  code: 'ERR_INVALID_ARG_TYPE'
}
```

First time after upgrading Next.js to 12.0.4.

### Expected Behavior

Server-side rendering to work.

### To Reproduce

Happened only after deployed to an EB environment and could reproduce locally from the exact artifacts generated on deploy time (so I'm afraid I may be the one to blame)

1. Build: `yarn install && yarn build`
2. Zip build before uploading: `mkdir dist && zip dist/***.zip -r dist .next package.json yarn.lock next.config.js public`
3. From the generated zip, `npm install && npm run start`

**Comment by jackyef** [comment]
I am facing a similar error. Curious, do you have a custom `_document` in your next app? If you do, how does it look?

**Comment by futurepastori** [comment]
@jackyef here's my `_document`: https://gist.github.com/futurepastori/960dca14ab7d56a63edfe582eab4b35b

And `package-lock.json` for refernece: https://gist.github.com/futurepastori/1d425250b8eae7cd1c9473645da65751

**Comment by camjw** [comment]
I also am getting this error on pages that have `getServerSideProps` methods. Specifically, we have an auth flow where we redirect `/` to `/auth/sign_in` which then has a `getServerSideProps` method. When we redirect from `/` to `/auth/sign_in` it works fine, but when I refresh the page on that route it then gives this error. The same thing happens on a couple of other routes. 

It only fails when running on elastic beanstalk, when I build the app and run it locally it works fine. 

The `_document` only adds random meta tags, there's no custom methods.

We're using next v12.0.0.

Hope this is useful!

**Comment by camjw** [comment]
Downgrading from next v12.0.0 to next v11.1.2 fixed this! There are quite a few google hits for this error and elastic beanstalk in a variety of contexts so I imagine it's really due to some quirk there.

**Comment by darwinshameran** [comment]
Having. the same issue whilst making a request to a serverless function on Vercel. We're running next v.12.0.4.

**Comment by mztnnrt** [comment]
I also have same error

**Comment by williamstein** [comment]
Summary: I built with 12.0.3 and served with 12.0.4, and that caused this problem for me.  Upgrading my build environment to 12.0.4 fixes the issue.  Details below.

I have the exact same issue when attempting to use a production build.  I'm not using a custom _document, but  my pages all have a custom getServerSideProps method.

I built everything using next 12.0.3, but I'm running the build artifacts using next 12.0.4, because of semver and my build pipeline.  This is totally broken.

Exactly the same code built and worked fine with next 12.0.3 both for building and for serving my nextjs site.    Thus I can try two things:
 - upgrade my build to use 12.0.4: I tried this and it **does work**.
 - downgrade my production site to use 12.0.3.  This would of course work, since that's exactly what I was doing last week.

It seems like there is a serious semver mistake in tagging 12.0.4 via semver as being compatible with 12.0.3.  Technically, I guess it should be 13.0.0 :-).

**Comment by Webbanditten** [comment]
We have the same issue with our build where 12.0.4 and up is breaking our build. It seems like it will get patched in 12.0.8.

**Comment by balazsorban44** [comment]
PR #32247 might be related, which is part of `12.0.8-canary.3` and higher.

@futurepastori could you give `yarn add next@canary` a try?

**Comment by futurepastori** [comment]
@balazsorban44 might be! Tried and can't reproduce the error anymore, even after running prod server on versions lower than build target.

Closing, thanks!

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
